Index: src/main/java/org/gauravagrwl/financeData/controller/BookController.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/org/gauravagrwl/financeData/controller/BookController.java b/src/main/java/org/gauravagrwl/financeData/controller/BookController.java
new file mode 100644
--- /dev/null	(date 1712625174583)
+++ b/src/main/java/org/gauravagrwl/financeData/controller/BookController.java	(date 1712625174583)
@@ -0,0 +1,19 @@
+package org.gauravagrwl.financeData.controller;
+
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class BookController {
+
+    @PostMapping("/add")
+    public void addTweet() {
+
+    }
+
+    @GetMapping("/hello")
+    public String hello(){
+        return "Hello World!";
+    }
+}
Index: src/main/java/org/gauravagrwl/financeData/controller/ProfileController.java
===================================================================
diff --git a/src/main/java/org/gauravagrwl/financeData/controller/ProfileController.java b/src/main/java/org/gauravagrwl/financeData/controller/ProfileController.java
deleted file mode 100644
--- a/src/main/java/org/gauravagrwl/financeData/controller/ProfileController.java	
+++ /dev/null	
@@ -1,44 +0,0 @@
-package org.gauravagrwl.financeData.controller;
-
-import jakarta.validation.Valid;
-import org.gauravagrwl.financeData.model.documents.profile.ProfileDocument;
-import org.gauravagrwl.financeData.service.ProfileService;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-import org.springframework.http.ResponseEntity;
-import org.springframework.web.bind.annotation.*;
-
-@RequestMapping(value = "/userProfile")
-public class ProfileController {
-
-    ProfileService profileService;
-
-    Logger LOGGER = LoggerFactory.getLogger(ProfileController.class);
-
-
-    public ProfileController(ProfileService profileService) {
-        this.profileService = profileService;
-    }
-
-    @PostMapping(value = "/createProfile", consumes = "application/json", produces = "application/json")
-    ResponseEntity<String> createUserProfileData(@Valid @RequestBody ProfileDocument profile){
-        String profileId = profileService.createUserProfile(profile);
-        return ResponseEntity.ok(profileId);
-
-    }
-    @GetMapping(value = "/getProfile", produces = "application/json")
-    public ResponseEntity<ProfileDocument> getProfileData(@RequestParam String userName){
-        ProfileDocument document = profileService.getProfileDocumentByUserName(userName);
-        return ResponseEntity.ok(document);
-    }
-
-    @DeleteMapping(value = "/deleteProfile", produces = "application/json")
-    public ResponseEntity<String> deleteProfileData(@RequestParam String userName){
-        profileService.deleteUserProfile(userName);
-        return ResponseEntity.ok("User is removed.");
-    }
-
-
-
-
-}
Index: src/main/java/org/gauravagrwl/financeData/controller/AccountController.java
===================================================================
diff --git a/src/main/java/org/gauravagrwl/financeData/controller/AccountController.java b/src/main/java/org/gauravagrwl/financeData/controller/AccountController.java
deleted file mode 100644
--- a/src/main/java/org/gauravagrwl/financeData/controller/AccountController.java	
+++ /dev/null	
@@ -1,101 +0,0 @@
-package org.gauravagrwl.financeData.controller;
-
-import jakarta.validation.Valid;
-import org.gauravagrwl.financeData.model.documents.account.AccountDocument;
-import org.gauravagrwl.financeData.service.ProfileAccountService;
-import org.gauravagrwl.financeData.utility.FinanceDataHelper;
-import org.gauravagrwl.financeData.utility.InstitutionCategoryEnum;
-import org.gauravagrwl.financeData.utility.InstitutionTypeEnum;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-import org.springframework.http.ResponseEntity;
-import org.springframework.web.bind.annotation.*;
-
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-
-@RestController
-@RequestMapping("/profileAccount")
-public class AccountController {
-    ProfileAccountService accountService;
-
-    Logger LOGGER = LoggerFactory.getLogger(AccountController.class);
-
-    public AccountController(ProfileAccountService accountService) {
-        this.accountService = accountService;
-    }
-
-    @PostMapping(value = "/addAccount", consumes = "application/json", produces = "application/json")
-    ResponseEntity<String> addUserAccount(@RequestParam(required = true) String userName,
-                                                   AccountDocument accountDocument){
-
-        return ResponseEntity.ok("Account Added with Id: ");
-    }
-    @PostMapping(value = "/addAccounts", consumes = "application/json", produces = "application/json")
-    ResponseEntity<?> addUserAccounts(@RequestParam(required = true) String userName,
-                                                   List<AccountDocument> accountDocuments){
-        Map<String, String> result = new HashMap<>();
-        accountDocuments.forEach(accountDocument -> {
-            try {
-                String userAccountId = accountService.addUserAccount(accountDocument, userName);
-                result.put(accountDocument.getAccountNumber(), "Account Added with Id: " + userAccountId);
-            } catch (Exception e) {
-                result.put(accountDocument.getAccountNumber(), "Account already exist.");
-            }
-        });
-        return ResponseEntity.ok(result);
-    }
-
-    @GetMapping(value = "/getAccount",  produces = "application/json")
-    ResponseEntity<?> getUserAccount(@RequestParam(required = true) String userName,
-                                                             @RequestParam(required = true) String accountId){
-        AccountDocument accountDocument = accountService.getUserAccountDocument(accountId, userName);
-        return ResponseEntity.ok(accountDocument);
-    }
-
-    @GetMapping(value = "/getAccounts",  produces = "application/json")
-    ResponseEntity<?> getUserAccounts(@RequestParam(required = true) String userName,@RequestParam(name = "institutionCategory", required = false) InstitutionCategoryEnum instCategory,
-                                                              @RequestParam(name = "accountType", required = false) InstitutionTypeEnum accountType) {
-        List<AccountDocument> userAccounts = accountService.getUserAccounts(userName);
-        userAccounts.forEach(account -> account
-                .setAccountNumber(FinanceDataHelper.prependAccountNumber(account.getAccountNumber())));
-        return ResponseEntity.ok(userAccounts);
-    }
-
-    @DeleteMapping(value = "/deleteAccount")
-    ResponseEntity<String> deleteUserAccount(@RequestParam(required = true) String userName, @RequestParam(required = true)
-                                             String accountId){
-        AccountDocument accountDocument = accountService.deleteUserAccountDocument(accountId, userName);
-        return null;
-    }
-    @PatchMapping(value = "/toggleAccountStatus")
-    ResponseEntity<String> toggleAccountActiveStatus(@RequestParam(required = true) String userName,
-                                                     @RequestParam(required = true)
-                                                     String accountId){
-        accountService.toggleAccountActiveStatus(userName, accountId);
-        return ResponseEntity.ok("Account is toggled!");
-    }
-
-    @GetMapping(value = "/getInstitutionType")
-    ResponseEntity<List<String>> getAccountType(
-            @RequestParam(name = "userName", required = false) String userName,
-            @RequestParam(name = "accountCategory", required = false) String accountCategory) {
-
-        List<String> accountTypeList = new ArrayList<>();
-        for (InstitutionTypeEnum accountType : InstitutionTypeEnum.values()) {
-            accountTypeList.add(accountType.name());
-        }
-        return ResponseEntity.ok(accountTypeList);
-    }
-
-    @GetMapping(value = "/getInstitutionCategory")
-    ResponseEntity<List<String>> getAccountCategory(@RequestParam(name = "userName", required = false) String userName) {
-        List<String> accountCatList = new ArrayList<>();
-        for (InstitutionCategoryEnum accountCat : InstitutionCategoryEnum.values()) {
-            accountCatList.add(accountCat.getCategoryName());
-        }
-        return ResponseEntity.ok(accountCatList);
-    }
-}
Index: src/main/java/org/gauravagrwl/financeData/controller/UploadController.java
===================================================================
diff --git a/src/main/java/org/gauravagrwl/financeData/controller/UploadController.java b/src/main/java/org/gauravagrwl/financeData/controller/UploadController.java
deleted file mode 100644
--- a/src/main/java/org/gauravagrwl/financeData/controller/UploadController.java	
+++ /dev/null	
@@ -1,29 +0,0 @@
-package org.gauravagrwl.financeData.controller;
-
-import org.gauravagrwl.financeData.service.ProfileService;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-import org.springframework.http.ResponseEntity;
-import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RequestParam;
-import org.springframework.web.bind.annotation.RestController;
-import org.springframework.web.multipart.MultipartFile;
-
-import java.io.IOException;
-
-@RestController
-@RequestMapping(value = "/upload")
-public class UploadController {
-
-    Logger LOGGER = LoggerFactory.getLogger(UploadController.class);
-
-    @PostMapping("/uploadDocuments")
-    public ResponseEntity<String> uploadAccountDocuments(
-            @RequestParam(name = "userName", required = true) String userName,
-            @RequestParam(name = "accountId", required = true) String accountId,
-            @RequestParam(required = true, name = "file") MultipartFile file) throws IOException {
-        return null;
-    }
-
-}
Index: src/main/java/org/gauravagrwl/financeData/controller/ReportController.java
===================================================================
diff --git a/src/main/java/org/gauravagrwl/financeData/controller/ReportController.java b/src/main/java/org/gauravagrwl/financeData/controller/ReportController.java
deleted file mode 100644
--- a/src/main/java/org/gauravagrwl/financeData/controller/ReportController.java	
+++ /dev/null	
@@ -1,13 +0,0 @@
-package org.gauravagrwl.financeData.controller;
-
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.RestController;
-
-@RestController
-public class ReportController {
-
-    @GetMapping(value = "/getString")
-    void getString(){
-
-    }
-}
Index: src/main/java/org/gauravagrwl/financeData/controller/FinanceDataController.java
===================================================================
diff --git a/src/main/java/org/gauravagrwl/financeData/controller/FinanceDataController.java b/src/main/java/org/gauravagrwl/financeData/controller/FinanceDataController.java
deleted file mode 100644
--- a/src/main/java/org/gauravagrwl/financeData/controller/FinanceDataController.java	
+++ /dev/null	
@@ -1,39 +0,0 @@
-package org.gauravagrwl.financeData.controller;
-
-import org.gauravagrwl.financeData.service.FinancialDataService;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-import org.springframework.http.ResponseEntity;
-import org.springframework.web.bind.annotation.*;
-
-@RestController
-@RequestMapping("/appController")
-public class FinanceDataController {
-
-    private final FinancialDataService financialDataService;
-
-    public FinanceDataController(FinancialDataService financialDataService) {
-        this.financialDataService = financialDataService;
-    }
-
-    Logger LOGGER = LoggerFactory.getLogger(FinanceDataController.class);
-
-    @DeleteMapping(path = "/dropAllCollections")
-    public ResponseEntity<?>dropAllCollections(){
-        LOGGER.warn("Dropping all Collections");
-        financialDataService.dropAllCollection();
-        return ResponseEntity.ok(financialDataService.dropAllCollection());
-    }
-
-    @DeleteMapping(path = "/dropCollection")
-    public ResponseEntity<?>dropCollection(@RequestParam String collectionName) {
-        LOGGER.warn(collectionName + "Will be dropped.");
-
-        return ResponseEntity.ok(financialDataService.dropCollection(collectionName));
-    }
-
-    @GetMapping(path = "/getCollections")
-    public ResponseEntity<?>getCollection() {
-        return ResponseEntity.ok(financialDataService.getAllCollections());
-    }
-}
Index: src/main/java/org/gauravagrwl/financeData/controller/StatementController.java
===================================================================
diff --git a/src/main/java/org/gauravagrwl/financeData/controller/StatementController.java b/src/main/java/org/gauravagrwl/financeData/controller/StatementController.java
deleted file mode 100644
--- a/src/main/java/org/gauravagrwl/financeData/controller/StatementController.java	
+++ /dev/null	
@@ -1,13 +0,0 @@
-package org.gauravagrwl.financeData.controller;
-
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.RestController;
-
-@RestController
-public class StatementController {
-    @GetMapping(value = "/getStringOne")
-    void getStringOne(){
-
-    }
-
-}
Index: src/main/java/org/gauravagrwl/financeData/FinanceDataApplication.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/org/gauravagrwl/financeData/FinanceDataApplication.java b/src/main/java/org/gauravagrwl/financeData/FinanceDataApplication.java
--- a/src/main/java/org/gauravagrwl/financeData/FinanceDataApplication.java	
+++ b/src/main/java/org/gauravagrwl/financeData/FinanceDataApplication.java	(date 1712625174582)
@@ -4,17 +4,9 @@
 import io.swagger.v3.oas.annotations.info.Info;
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
-import org.springframework.context.annotation.Bean;
-import org.springframework.data.auditing.DateTimeProvider;
-import org.springframework.data.domain.AuditorAware;
-import org.springframework.data.mongodb.repository.config.EnableMongoRepositories;
-
-import java.time.OffsetDateTime;
-import java.util.Optional;
 
 @SpringBootApplication
-@OpenAPIDefinition(info =@Info(title = "Finance Data API", version = "1.0.0"))
-@EnableMongoRepositories(value = "org.gauravagrwl.financeData.model.*")
+@OpenAPIDefinition(info =@Info(title = "Finance Data API"))
 public class FinanceDataApplication {
 
 	public static void main(String[] args) {
@@ -22,12 +14,3 @@
 	}
 
 }
-
-
-//TODO: Naming Conventions
-//TODO: Java Docs
-//TODO: Add API Docs
-//TODO: Set user from SecurityContext
-//TODO: Set Security Context (JWT or LDAP or SAML or OAuth)
-//TODO: Exception Handling
-
Index: src/main/java/org/gauravagrwl/financeData/model/documents/account/AccountDocument.java
===================================================================
diff --git a/src/main/java/org/gauravagrwl/financeData/model/documents/account/AccountDocument.java b/src/main/java/org/gauravagrwl/financeData/model/documents/account/AccountDocument.java
deleted file mode 100644
--- a/src/main/java/org/gauravagrwl/financeData/model/documents/account/AccountDocument.java	
+++ /dev/null	
@@ -1,69 +0,0 @@
-package org.gauravagrwl.financeData.model.documents.account;
-
-import com.fasterxml.jackson.annotation.JsonCreator;
-import com.fasterxml.jackson.annotation.JsonSubTypes;
-import com.fasterxml.jackson.annotation.JsonTypeInfo;
-import jakarta.validation.constraints.NotBlank;
-import lombok.*;
-import org.gauravagrwl.financeData.model.documents.audit.AuditMetaData;
-import org.gauravagrwl.financeData.utility.InstitutionCategoryEnum;
-import org.gauravagrwl.financeData.utility.InstitutionTypeEnum;
-import org.springframework.data.annotation.Version;
-import org.springframework.data.mongodb.core.index.Indexed;
-import org.springframework.data.mongodb.core.mapping.MongoId;
-
-import java.io.Serializable;
-import java.math.BigDecimal;
-import java.util.Currency;
-import java.util.UUID;
-
-@AllArgsConstructor(onConstructor = @__(@JsonCreator))
-@NoArgsConstructor
-@Getter
-@Setter
-@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.EXISTING_PROPERTY, property = "institutionCategory", defaultImpl = InstitutionCategoryEnum.class, visible = true)
-@JsonSubTypes({
-        @JsonSubTypes.Type(value = BankAccountDocument.class, name = "BankAccount"),
-        @JsonSubTypes.Type(value = InvestmentAccountDocument.class, name = "InvestmentAccount"),
-        @JsonSubTypes.Type(value = LoanAccountDocument.class, name = "LoanAccount"),
-        @JsonSubTypes.Type(value = AssetsAccountDocument.class, name = "AssetsAccount"),
-})
-public abstract class AccountDocument {
-    @MongoId
-    private String id;
-
-    // Financial institute Name
-    @NotBlank(message = "Institute Name (IN / USA ) is mandatory. ")
-    private String institutionName;
-
-    // Financial institute Currency
-    private Currency institutionCurrency;
-
-    // Financial institute Type can be: BANKING, INVESTMENT, MARKET, LOAN
-    private InstitutionCategoryEnum institutionCategory;
-
-    // Financial institute Account Number must be unique
-    @Indexed(unique = true, background = true)
-    private String accountNumber;
-
-    private InstitutionTypeEnum accountType;
-
-    // User profile who holds the account
-    private String profileDocumentId;
-
-    // Is this account still Active.
-    private Boolean isActive = Boolean.TRUE;
-
-    //Set the CollectionName for statement
-    private String collectionName;
-
-    private AuditMetaData audit = new AuditMetaData();
-
-    @Version
-    private Integer version;
-
-    // Indicator if respective Account Balance Is Calculated.
-    private Boolean isBalanceCalculated = Boolean.FALSE;
-
-    public abstract void calculate(BigDecimal amount);
-}
Index: src/main/java/org/gauravagrwl/financeData/model/documents/account/LoanAccountDocument.java
===================================================================
diff --git a/src/main/java/org/gauravagrwl/financeData/model/documents/account/LoanAccountDocument.java b/src/main/java/org/gauravagrwl/financeData/model/documents/account/LoanAccountDocument.java
deleted file mode 100644
--- a/src/main/java/org/gauravagrwl/financeData/model/documents/account/LoanAccountDocument.java	
+++ /dev/null	
@@ -1,34 +0,0 @@
-package org.gauravagrwl.financeData.model.documents.account;
-
-import lombok.*;
-
-import java.math.BigDecimal;
-
-@AllArgsConstructor
-@NoArgsConstructor
-@Setter
-@Getter
-public class LoanAccountDocument extends AccountDocument {
-
-    private BigDecimal loanAmount;
-    private BigDecimal remaingAmount;
-    private BigDecimal amountPaid;
-    private String rateOfIntrest;
-    private String purpose;
-    private String remaingInstallment;
-
-    @Override
-    public void calculate(BigDecimal amount) {
-        // TODO Auto-generated method stub
-        throw new UnsupportedOperationException("Unimplemented method 'calculate'");
-    }
-
-//    @Override
-//    public MappingStrategy<? extends AccountStatementDocument> getMappingStrategy(String profileType) {
-//        MappingStrategy<BankAccountStatementDocument> headerColumnNameMappingStrategy = new HeaderColumnNameMappingStrategyBuilder<BankAccountStatementDocument>()
-//                .withForceCorrectRecordLength(true).build();
-//        headerColumnNameMappingStrategy.setProfile(profileType);
-//        headerColumnNameMappingStrategy.setType(BankAccountStatementDocument.class);
-//        return headerColumnNameMappingStrategy;
-//    }
-}
Index: src/main/java/org/gauravagrwl/financeData/model/documents/account/AssetsAccountDocument.java
===================================================================
diff --git a/src/main/java/org/gauravagrwl/financeData/model/documents/account/AssetsAccountDocument.java b/src/main/java/org/gauravagrwl/financeData/model/documents/account/AssetsAccountDocument.java
deleted file mode 100644
--- a/src/main/java/org/gauravagrwl/financeData/model/documents/account/AssetsAccountDocument.java	
+++ /dev/null	
@@ -1,44 +0,0 @@
-package org.gauravagrwl.financeData.model.documents.account;
-
-
-import lombok.*;
-
-import java.math.BigDecimal;
-
-@AllArgsConstructor
-@NoArgsConstructor
-@Setter
-@Getter
-public class AssetsAccountDocument extends AccountDocument {
-
-    // Total amount for the purchase.
-    private BigDecimal amountInvestment;
-
-    // Asset name
-    private String propertyName;
-
-    // Asset acquired Year
-    private String acquiredYear;
-
-    private boolean isRentable;
-
-    private String tenantName;
-
-    private String tenantNumber;
-
-    @Override
-    public void calculate(BigDecimal amount) {
-        // TODO Auto-generated method stub
-        throw new UnsupportedOperationException("Unimplemented method 'calculate'");
-    }
-
-//    @Override
-//    public MappingStrategy<? extends AccountStatementDocument> getMappingStrategy(String profileType) {
-//        MappingStrategy<BankAccountStatementDocument> headerColumnNameMappingStrategy = new HeaderColumnNameMappingStrategyBuilder<BankAccountStatementDocument>()
-//                .withForceCorrectRecordLength(true).build();
-//        headerColumnNameMappingStrategy.setProfile(profileType);
-//        headerColumnNameMappingStrategy.setType(BankAccountStatementDocument.class);
-//        return headerColumnNameMappingStrategy;
-//    }
-
-}
Index: src/main/java/org/gauravagrwl/financeData/model/documents/account/BankAccountDocument.java
===================================================================
diff --git a/src/main/java/org/gauravagrwl/financeData/model/documents/account/BankAccountDocument.java b/src/main/java/org/gauravagrwl/financeData/model/documents/account/BankAccountDocument.java
deleted file mode 100644
--- a/src/main/java/org/gauravagrwl/financeData/model/documents/account/BankAccountDocument.java	
+++ /dev/null	
@@ -1,49 +0,0 @@
-package org.gauravagrwl.financeData.model.documents.account;
-
-import lombok.*;
-
-import java.math.BigDecimal;
-
-@AllArgsConstructor
-@NoArgsConstructor
-@Setter
-@Getter
-public class BankAccountDocument extends AccountDocument {
-
-    // Account Calculated Balance
-    private BigDecimal accountBalance = BigDecimal.ZERO;
-
-    // Account holding type
-    private String holdingType;
-
-    // Account Code: Routing code or IIFC code.
-    private String accountCode;
-
-    // Code Type: Routing or IIFC
-    private String accountCodeType;
-
-    @Override
-    public void calculate(BigDecimal amount) {
-        this.accountBalance = amount;
-    }
-
-//    @Override
-//    public MappingStrategy<? extends AccountStatementDocument> getMappingStrategy(String profileType) {
-//        MappingStrategy<BankAccountStatementDocument> headerColumnNameMappingStrategy = new HeaderColumnNameMappingStrategyBuilder<BankAccountStatementDocument>()
-//                .withForceCorrectRecordLength(true).build();
-//        headerColumnNameMappingStrategy.setProfile(profileType);
-//        headerColumnNameMappingStrategy.setType(BankAccountStatementDocument.class);
-//        return headerColumnNameMappingStrategy;
-//    }
-
-//    @Override
-//    public MappingStrategy<BankAccountStatementDocument> getMappingStrategy(String profileType) {
-//        MappingStrategy<BankAccountStatementDocument> headerColumnNameMappingStrategy = new HeaderColumnNameMappingStrategyBuilder<BankAccountStatementDocument>()
-//                .withForceCorrectRecordLength(true).build();
-//        headerColumnNameMappingStrategy.setProfile(profileType);
-//        headerColumnNameMappingStrategy.setType(BankAccountStatementDocument.class);
-//        return headerColumnNameMappingStrategy;
-//    }
-
-
-}
\ No newline at end of file
Index: src/main/java/org/gauravagrwl/financeData/model/documents/account/InvestmentAccountDocument.java
===================================================================
diff --git a/src/main/java/org/gauravagrwl/financeData/model/documents/account/InvestmentAccountDocument.java b/src/main/java/org/gauravagrwl/financeData/model/documents/account/InvestmentAccountDocument.java
deleted file mode 100644
--- a/src/main/java/org/gauravagrwl/financeData/model/documents/account/InvestmentAccountDocument.java	
+++ /dev/null	
@@ -1,39 +0,0 @@
-package org.gauravagrwl.financeData.model.documents.account;
-
-import lombok.*;
-
-import java.math.BigDecimal;
-
-@AllArgsConstructor
-@NoArgsConstructor
-@Setter
-@Getter
-public class InvestmentAccountDocument extends AccountDocument {
-
-    // Total Amount Invested.
-    private BigDecimal amountInvestment = BigDecimal.ZERO;
-
-    // Total Amount Returned.
-    private BigDecimal amountReturn = BigDecimal.ZERO;
-
-    // Is this Account Auto Tradeable.
-    private Boolean isAutoTradable = Boolean.FALSE;
-
-    private String ledgerTransactionCollectionName;
-
-    @Override
-    public void calculate(BigDecimal amount) {
-        // TODO Auto-generated method stub
-        throw new UnsupportedOperationException("Unimplemented method 'calculate'");
-    }
-
-//    @Override
-//    public MappingStrategy<InvestmentAccountStatement> getMappingStrategy(String profileType) {
-//        MappingStrategy<InvestmentAccountStatement> headerColumnNameMappingStrategy = new HeaderColumnNameMappingStrategyBuilder<InvestmentAccountStatement>()
-//                .withForceCorrectRecordLength(true).build();
-//        headerColumnNameMappingStrategy.setProfile(profileType);
-//        headerColumnNameMappingStrategy.setType(InvestmentAccountStatement.class);
-//        return headerColumnNameMappingStrategy;
-//    }
-
-}
Index: src/main/java/org/gauravagrwl/financeData/model/documents/profile/ProfileDocument.java
===================================================================
diff --git a/src/main/java/org/gauravagrwl/financeData/model/documents/profile/ProfileDocument.java b/src/main/java/org/gauravagrwl/financeData/model/documents/profile/ProfileDocument.java
deleted file mode 100644
--- a/src/main/java/org/gauravagrwl/financeData/model/documents/profile/ProfileDocument.java	
+++ /dev/null	
@@ -1,42 +0,0 @@
-package org.gauravagrwl.financeData.model.documents.profile;
-
-import jakarta.validation.constraints.NotBlank;
-import lombok.Data;
-import org.gauravagrwl.financeData.model.documents.audit.AuditMetaData;
-import org.springframework.data.annotation.Version;
-import org.springframework.data.mongodb.core.index.Indexed;
-import org.springframework.data.mongodb.core.mapping.Document;
-import org.springframework.data.mongodb.core.mapping.MongoId;
-
-import java.util.UUID;
-
-@Data
-@Document(collection = "profileDocument")
-public class ProfileDocument {
-    @MongoId
-    private String id;
-
-    @NotBlank(message = "First Name is required.")
-    private String firstName;
-
-    @NotBlank(message = "Last Name is required.")
-    private String lastname;
-
-    private String secrectKey;
-
-    @NotBlank(message = "Email Address is required.")
-    private String emailAddress;
-
-    @NotBlank(message = "Phone Number is required.")
-    private String phoneNumber;
-
-    @NotBlank(message = "Phone Number is required.")
-    @Indexed(unique = true, background = true)
-    private String userName;
-
-    private AuditMetaData audit = new AuditMetaData();
-
-    @Version
-    private Integer version;
-
-}
Index: src/main/java/org/gauravagrwl/financeData/model/documents/audit/AuditMetaData.java
===================================================================
diff --git a/src/main/java/org/gauravagrwl/financeData/model/documents/audit/AuditMetaData.java b/src/main/java/org/gauravagrwl/financeData/model/documents/audit/AuditMetaData.java
deleted file mode 100644
--- a/src/main/java/org/gauravagrwl/financeData/model/documents/audit/AuditMetaData.java	
+++ /dev/null	
@@ -1,27 +0,0 @@
-package org.gauravagrwl.financeData.model.documents.audit;
-
-import lombok.*;
-import org.springframework.data.annotation.CreatedBy;
-import org.springframework.data.annotation.CreatedDate;
-import org.springframework.data.annotation.LastModifiedBy;
-import org.springframework.data.annotation.LastModifiedDate;
-
-import java.time.Instant;
-
-@AllArgsConstructor
-@NoArgsConstructor
-@Getter
-@Setter
-public class AuditMetaData {
-    @CreatedDate
-    private Instant createdDate;
-
-    @LastModifiedDate
-    private Instant modifiedDate;
-
-    @CreatedBy
-    private String createdBy;
-
-    @LastModifiedBy
-    private String modifyBy;
-}
Index: src/main/java/org/gauravagrwl/financeData/model/repository/AccountDocumentRepository.java
===================================================================
diff --git a/src/main/java/org/gauravagrwl/financeData/model/repository/AccountDocumentRepository.java b/src/main/java/org/gauravagrwl/financeData/model/repository/AccountDocumentRepository.java
deleted file mode 100644
--- a/src/main/java/org/gauravagrwl/financeData/model/repository/AccountDocumentRepository.java	
+++ /dev/null	
@@ -1,15 +0,0 @@
-package org.gauravagrwl.financeData.model.repository;
-
-import org.gauravagrwl.financeData.model.documents.account.AccountDocument;
-import org.springframework.data.domain.Sort;
-import org.springframework.data.mongodb.repository.MongoRepository;
-import org.springframework.stereotype.Repository;
-
-import java.util.List;
-
-@Repository
-public interface AccountDocumentRepository extends MongoRepository<AccountDocument, String> {
-    boolean existsByIdAndProfileDocumentId(String accountId, String profileId);
-
-    List<AccountDocument> findByProfileDocumentId(String profileId, Sort sort);
-}
Index: src/main/java/org/gauravagrwl/financeData/model/repository/ProfileDocumentRepository.java
===================================================================
diff --git a/src/main/java/org/gauravagrwl/financeData/model/repository/ProfileDocumentRepository.java b/src/main/java/org/gauravagrwl/financeData/model/repository/ProfileDocumentRepository.java
deleted file mode 100644
--- a/src/main/java/org/gauravagrwl/financeData/model/repository/ProfileDocumentRepository.java	
+++ /dev/null	
@@ -1,18 +0,0 @@
-package org.gauravagrwl.financeData.model.repository;
-
-import org.gauravagrwl.financeData.model.documents.profile.ProfileDocument;
-import org.springframework.data.mongodb.repository.MongoRepository;
-import org.springframework.stereotype.Repository;
-
-import java.util.List;
-
-@Repository
-public interface ProfileDocumentRepository extends MongoRepository<ProfileDocument, String> {
-
-    public boolean existsByUserName(String userName);
-
-    public List<ProfileDocument> findByUserName(String username);
-
-
-
-}
Index: src/main/java/org/gauravagrwl/financeData/config/FinanceDataConfig.java
===================================================================
diff --git a/src/main/java/org/gauravagrwl/financeData/config/FinanceDataConfig.java b/src/main/java/org/gauravagrwl/financeData/config/FinanceDataConfig.java
deleted file mode 100644
--- a/src/main/java/org/gauravagrwl/financeData/config/FinanceDataConfig.java	
+++ /dev/null	
@@ -1,66 +0,0 @@
-package org.gauravagrwl.financeData.config;
-
-import jakarta.validation.ConstraintViolationException;
-import org.springframework.boot.web.client.RestTemplateBuilder;
-import org.springframework.context.annotation.Bean;
-import org.springframework.context.annotation.Configuration;
-import org.springframework.data.auditing.DateTimeProvider;
-import org.springframework.data.domain.AuditorAware;
-import org.springframework.data.mongodb.config.EnableMongoAuditing;
-import org.springframework.data.mongodb.core.mapping.event.ValidatingMongoEventListener;
-import org.springframework.http.HttpStatus;
-import org.springframework.validation.FieldError;
-import org.springframework.validation.beanvalidation.LocalValidatorFactoryBean;
-import org.springframework.web.bind.MethodArgumentNotValidException;
-import org.springframework.web.bind.annotation.ExceptionHandler;
-import org.springframework.web.bind.annotation.ResponseStatus;
-import org.springframework.web.client.RestTemplate;
-
-import java.time.OffsetDateTime;
-import java.util.HashMap;
-import java.util.Map;
-import java.util.Optional;
-
-@Configuration
-@EnableMongoAuditing
-public class FinanceDataConfig {
-
-    @Bean
-    RestTemplate restTemplate(){
-        return new RestTemplateBuilder().build();
-    }
-
-    @Bean(name = "auditingDateTimeProvider")
-    DateTimeProvider dateTimeProvider() {
-        return () -> Optional.of(OffsetDateTime.now());
-    }
-
-    @Bean(name = "currentAuditor")
-    AuditorAware<String> getCurrentAuditor() {
-        return () -> Optional.of("home-app");
-    }
-
-    @Bean
-    public ValidatingMongoEventListener validatingMongoEventListener() {
-        return new ValidatingMongoEventListener(validator());
-    }
-
-    @Bean
-    public LocalValidatorFactoryBean validator() {
-        return new LocalValidatorFactoryBean();
-    }
-
-    @ResponseStatus(HttpStatus.BAD_REQUEST)
-    @ExceptionHandler(MethodArgumentNotValidException.class)
-    public Map<String, String> handleValidationExceptions(
-            MethodArgumentNotValidException ex) {
-        Map<String, String> errors = new HashMap<>();
-        ex.getBindingResult().getAllErrors().forEach((error) -> {
-            String fieldName = ((FieldError) error).getField();
-            String errorMessage = error.getDefaultMessage();
-            errors.put(fieldName, errorMessage);
-        });
-        return errors;
-    }
-
-}
Index: src/main/java/org/gauravagrwl/financeData/config/WebSecurityConfig.java
===================================================================
diff --git a/src/main/java/org/gauravagrwl/financeData/config/WebSecurityConfig.java b/src/main/java/org/gauravagrwl/financeData/config/WebSecurityConfig.java
deleted file mode 100644
--- a/src/main/java/org/gauravagrwl/financeData/config/WebSecurityConfig.java	
+++ /dev/null	
@@ -1,39 +0,0 @@
-package org.gauravagrwl.financeData.config;
-
-import org.springframework.context.annotation.Bean;
-import org.springframework.context.annotation.Configuration;
-import org.springframework.security.config.annotation.web.builders.HttpSecurity;
-import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
-import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer;
-import org.springframework.security.web.SecurityFilterChain;
-import org.springframework.web.cors.CorsConfiguration;
-import org.springframework.web.cors.CorsConfigurationSource;
-import org.springframework.web.cors.UrlBasedCorsConfigurationSource;
-
-import java.util.Arrays;
-
-@Configuration
-@EnableWebSecurity
-public class WebSecurityConfig {
-
-    @Bean
-    SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
-        http.authorizeHttpRequests(request ->
-                        request.anyRequest().permitAll())
-                .csrf(AbstractHttpConfigurer::disable)
-                .cors(cors ->
-                        cors.configurationSource(corsConfigurationSource()));
-        return http.build();
-    }
-
-    @Bean
-    CorsConfigurationSource corsConfigurationSource() {
-        CorsConfiguration configuration = new CorsConfiguration();
-        configuration.setAllowedOrigins(Arrays.asList("*"));
-        configuration.setAllowedMethods(Arrays.asList("*"));
-        configuration.setAllowedHeaders(Arrays.asList("*"));
-        UrlBasedCorsConfigurationSource source = new UrlBasedCorsConfigurationSource();
-        source.registerCorsConfiguration("/**", configuration);
-        return source;
-    }
-}
Index: src/main/java/org/gauravagrwl/financeData/utility/FinanceDataHelper.java
===================================================================
diff --git a/src/main/java/org/gauravagrwl/financeData/utility/FinanceDataHelper.java b/src/main/java/org/gauravagrwl/financeData/utility/FinanceDataHelper.java
deleted file mode 100644
--- a/src/main/java/org/gauravagrwl/financeData/utility/FinanceDataHelper.java	
+++ /dev/null	
@@ -1,19 +0,0 @@
-package org.gauravagrwl.financeData.utility;
-
-import org.apache.commons.lang3.StringUtils;
-
-public class FinanceDataHelper {
-    public static final String UNDER_SCORE = "_";
-
-    public static String prependAccountNumber(String accountNumber) {
-        return StringUtils.join("XXXX", StringUtils.right(accountNumber, 4));
-    }
-
-    public static String getStatementCollectionName(String... elements) {
-        return StringUtils.join(elements, UNDER_SCORE).concat("_statementCollection");
-    }
-
-    public static String getLedgerCollectionName(String... elements) {
-        return StringUtils.join(elements, UNDER_SCORE).concat("_ledgerDocument");
-    }
-}
Index: src/main/java/org/gauravagrwl/financeData/utility/InstitutionTypeEnum.java
===================================================================
diff --git a/src/main/java/org/gauravagrwl/financeData/utility/InstitutionTypeEnum.java b/src/main/java/org/gauravagrwl/financeData/utility/InstitutionTypeEnum.java
deleted file mode 100644
--- a/src/main/java/org/gauravagrwl/financeData/utility/InstitutionTypeEnum.java	
+++ /dev/null	
@@ -1,52 +0,0 @@
-package org.gauravagrwl.financeData.utility;
-
-import java.util.EnumSet;
-
-public enum InstitutionTypeEnum {
-
-    CHECKING(InstitutionCategoryEnum.BANKING, "CHK", "101"),
-    SAVING(InstitutionCategoryEnum.BANKING, "SAV", "102"),
-    DEPOSIT(InstitutionCategoryEnum.BANKING, "DEP", "103"),
-    PPF(InstitutionCategoryEnum.BANKING, "PPF", "104"),
-    CREDIT(InstitutionCategoryEnum.BANKING, "CRE", "105"),
-
-    STOCK(InstitutionCategoryEnum.INVESTMENT, "STOCK", "201"),
-    CRYPTO(InstitutionCategoryEnum.INVESTMENT, "CRYPTO", "202"),
-
-    PROPERTY(InstitutionCategoryEnum.ASSETS, "PROPERTY", "301"),
-
-    LOAN(InstitutionCategoryEnum.LOAN, "LOAN", "401"),
-    ;
-
-    private final InstitutionCategoryEnum accountCategory;
-    private final String accountTypeName, accountTypeCode;
-
-    /**
-     *
-     * @param accountCategory
-     * @param accountTypeName
-     * @param accountTypeCode
-     */
-    private InstitutionTypeEnum(InstitutionCategoryEnum accountCategory, String accountTypeName, String accountTypeCode) {
-        this.accountCategory = accountCategory;
-        this.accountTypeName = accountTypeName;
-        this.accountTypeCode = accountTypeCode;
-    }
-
-    public InstitutionCategoryEnum getAccountCategory() {
-        return accountCategory;
-    }
-
-    public String getAccountTypeName() {
-        return accountTypeName;
-    }
-
-    public String getAccountTypeCode() {
-        return accountTypeCode;
-    }
-
-    public EnumSet<InstitutionTypeEnum> BankingEnumSet() {
-        return EnumSet.of(InstitutionTypeEnum.CHECKING, InstitutionTypeEnum.SAVING, InstitutionTypeEnum.DEPOSIT,
-                InstitutionTypeEnum.PPF, InstitutionTypeEnum.CREDIT);
-    }
-}
Index: src/main/java/org/gauravagrwl/financeData/utility/InstitutionCategoryEnum.java
===================================================================
diff --git a/src/main/java/org/gauravagrwl/financeData/utility/InstitutionCategoryEnum.java b/src/main/java/org/gauravagrwl/financeData/utility/InstitutionCategoryEnum.java
deleted file mode 100644
--- a/src/main/java/org/gauravagrwl/financeData/utility/InstitutionCategoryEnum.java	
+++ /dev/null	
@@ -1,29 +0,0 @@
-package org.gauravagrwl.financeData.utility;
-
-public enum InstitutionCategoryEnum {
-    BANKING("BankAccount", "1"),
-    INVESTMENT("InvestmentAccount", "2"),
-    ASSETS("AssetsAccount", "3"),
-    LOAN("LoanAccount", "4"),
-    ;
-
-    private String categoryName;
-    private String categoryCode;
-
-    private InstitutionCategoryEnum(String categoryName, String categoryCode) {
-        this.categoryName = categoryName;
-        this.categoryCode = categoryCode;
-    }
-
-    private InstitutionCategoryEnum(String categoryName, Integer categoryCode) {
-        this(categoryName, categoryCode.toString());
-    }
-
-    public String getCategoryName() {
-        return categoryName;
-    }
-
-    public String getCategoryCode() {
-        return categoryCode;
-    }
-}
Index: src/main/java/org/gauravagrwl/financeData/service/ProfileService.java
===================================================================
diff --git a/src/main/java/org/gauravagrwl/financeData/service/ProfileService.java b/src/main/java/org/gauravagrwl/financeData/service/ProfileService.java
deleted file mode 100644
--- a/src/main/java/org/gauravagrwl/financeData/service/ProfileService.java	
+++ /dev/null	
@@ -1,43 +0,0 @@
-package org.gauravagrwl.financeData.service;
-
-import org.gauravagrwl.financeData.controller.AccountController;
-import org.gauravagrwl.financeData.exception.FinanceDataException;
-import org.gauravagrwl.financeData.model.documents.profile.ProfileDocument;
-import org.gauravagrwl.financeData.model.repository.ProfileDocumentRepository;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-import org.springframework.stereotype.Service;
-
-import java.util.List;
-
-@Service
-public class ProfileService {
-
-    ProfileDocumentRepository repo;
-
-    Logger LOGGER = LoggerFactory.getLogger(ProfileService.class);
-
-
-    public ProfileService(ProfileDocumentRepository repo) {
-        this.repo = repo;
-    }
-
-    public String createUserProfile(ProfileDocument profileDocument) {
-        if (repo.existsByUserName(profileDocument.getUserName()))
-            throw new FinanceDataException("User already exist!");
-        ProfileDocument insert = repo.insert(profileDocument);
-        return ("Document Inserted with id: " + insert.getId());
-    }
-
-    public void deleteUserProfile(String userName) {
-
-    }
-
-    public ProfileDocument getProfileDocumentByUserName(String userName) {
-        List<ProfileDocument> userDocuments = repo.findByUserName(userName);
-        if (userDocuments.size() != 1) {
-            throw new FinanceDataException("No user found!");
-        }
-        return userDocuments.get(0);
-    }
-}
Index: src/main/java/org/gauravagrwl/financeData/service/FinancialDataService.java
===================================================================
diff --git a/src/main/java/org/gauravagrwl/financeData/service/FinancialDataService.java b/src/main/java/org/gauravagrwl/financeData/service/FinancialDataService.java
deleted file mode 100644
--- a/src/main/java/org/gauravagrwl/financeData/service/FinancialDataService.java	
+++ /dev/null	
@@ -1,35 +0,0 @@
-package org.gauravagrwl.financeData.service;
-
-import org.apache.commons.lang3.StringUtils;
-import org.springframework.data.mongodb.core.MongoTemplate;
-import org.springframework.stereotype.Service;
-
-import java.util.Set;
-
-@Service
-public class FinancialDataService {
-    MongoTemplate template;
-
-    public FinancialDataService(MongoTemplate template) {
-        this.template = template;
-    }
-
-    public String dropAllCollection(){
-        template.getCollectionNames().stream().forEach(col -> template.dropCollection(col));
-        return "Warning All Database is dropped";
-    }
-
-    public String dropCollection(String collectionName){
-        template.getCollectionNames().stream().forEach(
-                col -> {
-                    if(StringUtils.equalsIgnoreCase(collectionName, col))
-                        template.dropCollection(col);
-                }
-        );
-        return collectionName + " is dropped";
-    }
-
-    public Set<String> getAllCollections() {
-        return template.getCollectionNames();
-    }
-}
Index: src/main/java/org/gauravagrwl/financeData/service/ProfileAccountService.java
===================================================================
diff --git a/src/main/java/org/gauravagrwl/financeData/service/ProfileAccountService.java b/src/main/java/org/gauravagrwl/financeData/service/ProfileAccountService.java
deleted file mode 100644
--- a/src/main/java/org/gauravagrwl/financeData/service/ProfileAccountService.java	
+++ /dev/null	
@@ -1,76 +0,0 @@
-package org.gauravagrwl.financeData.service;
-
-import org.gauravagrwl.financeData.controller.AccountController;
-import org.gauravagrwl.financeData.exception.FinanceDataException;
-import org.gauravagrwl.financeData.model.documents.account.AccountDocument;
-import org.gauravagrwl.financeData.model.repository.AccountDocumentRepository;
-import org.gauravagrwl.financeData.utility.FinanceDataHelper;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-import org.springframework.data.domain.Sort;
-import org.springframework.stereotype.Service;
-
-import java.util.List;
-
-@Service
-public class ProfileAccountService {
-
-    AccountDocumentRepository accountDocumentRepository;
-
-    Logger LOGGER = LoggerFactory.getLogger(ProfileAccountService.class);
-
-    ProfileService profileService;
-
-    public ProfileAccountService(AccountDocumentRepository accountDocumentRepository, ProfileService profileService) {
-        this.accountDocumentRepository = accountDocumentRepository;
-        this.profileService = profileService;
-    }
-
-    public String addUserAccount(AccountDocument accountDocument, String userName) {
-        String profileId = profileService.getProfileDocumentByUserName(userName).getId();
-        accountDocument.setProfileDocumentId(profileId);
-
-        accountDocument.setCollectionName(
-                FinanceDataHelper.getStatementCollectionName(accountDocument.getAccountNumber())
-        );
-        AccountDocument save = accountDocumentRepository.save(accountDocument);
-
-        return save.getId();
-    }
-
-    public AccountDocument getUserAccountDocument(String accountId, String userName) {
-        if (accountId != null && isUserAccountExist(accountId, userName)) {
-            return accountDocumentRepository.findById(accountId).get();
-        }
-        throw new FinanceDataException("Account Id is null or not exist.");
-    }
-
-    public boolean isUserAccountExist(String accountId, String userName) {
-        String profileId = profileService.getProfileDocumentByUserName(userName).getId();
-        return accountDocumentRepository.existsByIdAndProfileDocumentId(accountId,
-                profileId);
-    }
-
-    public List<AccountDocument> getUserAccounts(String userName) {
-        String profileId = profileService.getProfileDocumentByUserName(userName).getId();
-        Sort sort = Sort.by(Sort.Direction.DESC, "institutionCurrency");
-        List<AccountDocument> byProfileDocumentId = accountDocumentRepository.findByProfileDocumentId(profileId, sort);
-        return byProfileDocumentId;
-    }
-
-    public AccountDocument deleteUserAccountDocument(String accountId, String userName) {
-        if (accountId != null && isUserAccountExist(accountId, userName)) {
-            accountDocumentRepository.deleteById(accountId);
-        }
-        throw new FinanceDataException("Account Id is null or not exist.");
-    }
-
-    public void toggleAccountActiveStatus(String userName, String accountId) {
-        if (accountId != null && isUserAccountExist(accountId, userName)) {
-            AccountDocument document = accountDocumentRepository.findById(accountId).get();
-            document.setIsActive(!document.getIsActive());
-            accountDocumentRepository.save(document);
-        }
-        throw new FinanceDataException("Account Id is null or not exist.");
-    }
-}
Index: src/main/java/org/gauravagrwl/financeData/exception/FinanceDataException.java
===================================================================
diff --git a/src/main/java/org/gauravagrwl/financeData/exception/FinanceDataException.java b/src/main/java/org/gauravagrwl/financeData/exception/FinanceDataException.java
deleted file mode 100644
--- a/src/main/java/org/gauravagrwl/financeData/exception/FinanceDataException.java	
+++ /dev/null	
@@ -1,8 +0,0 @@
-package org.gauravagrwl.financeData.exception;
-
-public class FinanceDataException extends RuntimeException{
-
-    public FinanceDataException(String message){
-        super(message);
-    }
-}
